[6parkbbs] added website support (#3191)

dodieboy opened the pull request:
Added support for both club.6parkbbs and web.6parkbbs.
Also added filter so only manga is download from the site


<!-- Issuehunt content -->

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#3191: [Site Request] 6parkbbs](https://issuehunt.io/repos/95580598/issues/3191)
---
</details>
<!-- /Issuehunt content-->

dodieboy committed:
[sixparkbbsweb] added website support (#3191)

dodieboy committed:
[sixparkbbsclub] added website support (#3191)

- Added sixparkbbs template which can be use for both club and web sub forum
- Added sixparkbbsclub support

27136773 commented on src/web/mjs/connectors/templates/SixParkbbs.mjs:
```diff
@@ -0,0 +1,74 @@
+
+    async _getChapters(manga) {
+        let chapterList = [null]
+        chapterList = chapterList.map(element =>{
```
May return without intermediate local variable
`return chapterList.map(...);`

dodieboy replied:
Tried it but does not work, there will error of `chapterList is not defined`

27136773 replied:
Ah sorry i meant
```javascript
return [{
    id: manga.id,
    title: manga.title
}];
```

dodieboy replied:
thks, this work

27136773 commented on src/web/mjs/connectors/templates/SixParkbbs.mjs:
```diff
@@ -0,0 +1,74 @@
+        return data.map(element => {
+            if(element.text.match(this.queryMangasMatch)){
+                return {
+                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
```
May use `this.url` instead of `request.url`

27136773 commented on src/web/mjs/connectors/templates/SixParkbbs.mjs:
```diff
@@ -0,0 +1,74 @@
+    async _getMangaFromURI(uri) {
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangaTitle);
+        let id = this.getRootRelativeOrAbsoluteLink(uri, request.url);
```
May use `this.url` instead of `request.url`

27136773 commented on src/web/mjs/connectors/templates/SixParkbbs.mjs:
```diff
@@ -0,0 +1,74 @@
+    async _getMangasFromPage(page) {
+        let request = new Request(new URL(this.path.replace('%PAGE%', `${page}`), this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
```
Should use `.filter` before `.map` and remove elements that does not meet the condition, to avoid `undefined` entries in the manga list

27136773 commented on src/web/mjs/connectors/templates/SixParkbbs.mjs:
```diff
@@ -0,0 +1,74 @@
+    async _getPages(chapter) {
+        let request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryPage);
+        return data.map(element => this.getAbsolutePath(element.attributes[0].value, request.url));
```
Is there a more reliable way instead of `.attributes[0]` (e.g. get attribute by name)?

dodieboy replied:
i can use .attributes.mydatasrc too

27136773 reviewed (changes requested):
Lint Errors needs to be fixed.
The manga list should be filtered.
Otherwise looks good, some optional improvements are possible...

Tested locally with random mangas and worked as expected

dodieboy committed:
[sixparkbbs] minor changes

- added this.sub to suppot sixparkbbs sub forum
- better mange list filter
- minor change stated by ronny1982
- minor improvement for _getPages()

dodieboy commented on src/web/mjs/connectors/templates/SixParkbbs.mjs:
```diff
@@ -22,53 +23,46 @@ export default class SixParkbbs extends Connector {
+        let request = new Request(new URL(this.sub + chapter.id, this.url), this.requestOptions);
         let data = await this.fetchDOM(request, this.queryPage);
-        return data.map(element => this.getAbsolutePath(element.attributes[0].value, request.url));
+        return data.map(element => this.getAbsolutePath(element.attributes.mydatasrc.value || element.attributes.src.value, request.url));
```
I added `|| element.attributes.src.value` because some manga do not have `element.attributes.mydatasrc.value` but still getting `Cannot read property 'value' of undefined`, anyone have any idea where I did wrong?

happen on "Raw Hero 第38+39话[连载]" manga for SixParkbbsClub connector

27136773 replied:
Try `element.getAttribute('mydatasrc') || element`, the `getAbsolutePath` automatically detect if it is an image with the `src` attribute and use it if available

dodieboy replied:
nice, this work

dodieboy committed:
[sixparkbbs] fix lint

dodieboy committed:
[sixparkbbs] minor changes

- minor changes to support some manga page format

27136773 commented on src/web/mjs/connectors/templates/SixParkbbs.mjs:
```diff
@@ -0,0 +1,68 @@
+        let request = new Request(new URL(this.sub + this.path.replace('%PAGE%', `${page}`), this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangas);
+        return data.filter(element=> {
+            return element.text.match(this.queryMangasMatch) != null;
```
This may fail when `this.queryMangasMatch` is not set, but i would expect that it would return `true` in this case, e.g.
```javascript
return !this.queryMangasMatch || this.queryMangasMatch.test(element.text);
```

dodieboy committed:
[sixparkbbs] minor improvement

- improvment for some manga page format
- minor change stated by ronny1982

27136773 reviewed (approved):
Looks good
